joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/TypeInferencePass.scala

```diff
@@ -0,0 +1,47 @@
+package io.joern.javasrc2cpg.passes
+
+import io.joern.javasrc2cpg.util.TypeInfoCalculator.TypeConstants
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.Call.PropertyNames
+import io.shiftleft.codepropertygraph.generated.nodes.Call
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+class TypeInferencePass(cpg: Cpg) extends ConcurrentWriterCpgPass[Call](cpg) {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def generateParts(): Array[Call] = {
+    cpg.call.methodFullName(s".*${TypeConstants.UnresolvedType}.*").toArray
+  }
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, call: Call): Unit = {
+    val candidateMethods = cpg.method.internal.nameExact(call.name).l
+
+    candidateMethods match {
+      case Nil           => // Call is to an unresolved external method, so we can't get any more information here.
+      case method :: Nil =>
+        // This is probably the correct method if the number of arguments is correct.
+        if (method.parameter.size == call.argument.size) {
+          diffGraph.setNodeProperty(call, PropertyNames.MethodFullName, method.fullName)
+          diffGraph.setNodeProperty(call, PropertyNames.Signature, method.signature)
```

Comment:
What are the values of `dispatchType` and `typeFullName` for these calls? Do we want to also override them?

Reply:
Good question! `typeFullName` should also be overridden, but I'm not sure about `dispatchType`. I'll look into that.